ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml

```diff
@@ -215,19 +228,31 @@ QObject {
     }
 
     function __processAccountBalance(address) {
+        if (!root.fromToken) {
+            return null
+        }
+
         let network = ModelUtils.getByKey(root.filteredFlatNetworksModel, "chainId", root.swapFormData.selectedNetworkChainId)
-        if(!!network) {
-            let balancesModel = ModelUtils.getByKey(filteredBalancesModel, "tokensKey", root.swapFormData.fromTokensKey, "balances")
-            let accountBalance = ModelUtils.getByKey(balancesModel, "account", address)
-            if(!accountBalance) {
-                return {
-                    balance: "0",
-                    iconUrl: network.iconUrl,
-                    chainColor: network.chainColor}
-            }
+
+        if (!network) {
```

Comment:
This is quite visible in the PR's video: this code is totally ignoring the `network` in the code below, and calculates the balance of the selected account across all networks; not your code I know, but worth fixing imo (CC @Khushboo-dev-cpp)

Reply:
Yeah, I'll rework this in my followup PR for the new `TokenSelector`, nvm